decryptor/base/decryptor.go

```diff
@@ -97,7 +97,7 @@ type DataDecryptor interface {
 	// acrastruct using secure message
 	// return decrypted data or data as is if fail
 	// db specific
-	ReadSymmetricKey(*keys.PrivateKey, io.Reader) ([]byte, []byte, error)
+	ReadSymmetricKey([]*keys.PrivateKey, io.Reader) ([]byte, []byte, error)
```

Comment:
probably it will cause compilation errors in acra-ee due to using this interface from acra-ce which changed here...
better to add new instead of changing existing. but if do so then we should provide new extended interface to not cause same compilation errors in acra-ee and use it here instead old base DataDecryptor...
or change it here and don't forget to change in acra-ee....

Reply:
Hm... Makes sense.

For `KeyStore` interface we have kept around the old interface that retrieves the latest private key. It should not be used now but it may still be necessary for Acra EE until keystore v2 support is added there.

I seemed to think that decryption is implementation detail of Acra CE which does not leak to Acra EE so it's possible to change the interface without compatibility considerations. However, now I've looked at the actual code and see that Acra EE has additional decryptors.

We'd better introduce new interfaces that support multiple keys alongside with single-keyed ones.

Reply:
After thinking a bit about it, I have decided that it will be easier to make it a flag day than bother with strict backwards compatibility. I have retained meaning of the old interface methods, only adding new ones. However, this still changes the interface definition, so any Acra EE code that implements this interface (and we have some copypasta there) will break. This is intentional as it makes it trivial to track which places in Acra EE need to be updated. 

However, once merged into Acra CE's `master` this will also break any long-running branches in Acra EE which base on older Acra EE's `master`. These branches will have to be rebased onto the latest Acra EE master after the merge is complete. Unless they change something in `Decryptor` usage, this should be a trivial conflict-less rebase.

An alternative is to go full ham on compatibility, introduce new methods in new `DecryptorEx` interface, teach Acra CE to work with both of them, eventually teach Acra EE to work with both of them too, and then almost immediately remove them. This will allow us to not bother people working on long-running outdated branches but has no other benefit once those branches are merged into master.

Since we strive to practice continuous integration instead of having multiple branches, I believe it will be easier to tell the developers to quickly rebase and carry on instead of going out of the way with compatibility shims to not bother them completely.